### PR TITLE
Tree-sitter rolling fixes, 1.117 edition

### PR DESCRIPTION
**Pull request by savetheclocktower**
Getting a head start on this one.

This PR starts by adding a modern Tree-sitter grammar for SCSS. For ages, there wasn't a grammar out there that had good enough support for SCSS's features, but I forked [this one](https://github.com/serenadeai/tree-sitter-scss/) a while back and very gradually [fixed the things that needed to be fixed](https://github.com/savetheclocktower/tree-sitter-scss). I think it's good enough now to bundle and ship and see if there are any edge cases that need to be addressed.

A few days ago, one of the main Tree-sitter contributors [made this parser](https://github.com/tree-sitter-grammars/tree-sitter-scss), and I could see us migrating to it in the future (I don't love being a parser maintainer myself)… but it's not there yet. If it ever passes the tests that exist in my fork, then it'd be worth switching to.

### CODE REVIEW

**Comment by DeeDeeG** [comment]
Is it worth reaching out to the "one of the main Tree-sitter contributors" and be so bold as to ask if they'd like to adopt your parser, or combine efforts until there's one parser that meets _both_ of your respective sets of expectations? (I dunno if this suggestion is a little "out there", but it came across my mind.)

Anyway, good to see language support expanding for modern tree-sitter.

**Comment by savetheclocktower** [comment]
> Is it worth reaching out to the "one of the main Tree-sitter contributors" and be so bold as to ask if they'd like to adopt your parser, or combine efforts until there's one parser that meets _both_ of your respective sets of expectations? (I dunno if this suggestion is a little "out there", but it came across my mind.)

No, it's a fine suggestion. He knows about my parser, but he didn't see it until after he'd made his own. I told him about my test corpus and I expect him to adopt it at some point, but I also trust his solutions to some problems more than I trust my own (which are sometimes hacky) and his parser “extends” the Tree-sitter CSS parser more formally than mine does, so I think his is a better foundation to start from.

For now, though, it's not ready.

**Comment by savetheclocktower** [comment]
Taking this out of draft so we have time to get it reviewed before 1.117.

File: packages/language-typescript/grammars/common/highlights.scm [hunk 1]
```diff
@@ -902,6 +903,20 @@
 "." @keyword.operator.accessor._LANG_
 "?." @keyword.operator.accessor.optional-chaining._LANG_
 
+; Optional chaining is illegal…
```
Comment (DeeDeeG) [inline packages/language-typescript/grammars/common/highlights.scm:906]: Unsure of the factuality of this comment, perhaps

```suggestion
; Optional chaining should be illegal…
```

/s _(as in, just kidding!)_

--- 

Ah, I understand now, from reading the corresponding comment fragments below! So, there are _specific contexts_ in which optional chaining is illegal. Got it. _Definitely never mind this, then..._

File: packages/language-typescript/grammars/common/highlights.scm [hunk 2]
```diff
@@ -837,26 +837,28 @@
 
 "=" @keyword.operator.assignment._LANG_
 (non_null_expression "!" @keyword.operator.non-null._LANG_)
-(unary_expression"!" @keyword.operator.unary._LANG_)
+(unary_expression "!" @keyword.operator.unary._LANG_)
 
 [
+  "&&="
```
Comment (DeeDeeG) [inline packages/language-typescript/grammars/common/highlights.scm:843]: Today I Learned...

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND_assignment

File: packages/language-typescript/grammars/common/highlights.scm [hunk 3]
```diff
@@ -837,26 +837,28 @@
...
+  "||="
+  "??="
   "+="
   "-="
   "*="
+  "**="
```
Comment (DeeDeeG) [inline packages/language-typescript/grammars/common/highlights.scm:849]: Another one: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Exponentiation_assignment

**Review by DeeDeeG — APPROVED** [review]
Not a huge one this time, especially if looking at it commit-by-commit.

I am largely going by faith, since I don't know a ton of the details of the scopes/grammars myself. But this _looks_ okay! FWIW that I say that!

Thanks as always for maintaining the grammars!

---

> [[language-gfm] Make each block-level HTML tag its own injection…](https://github.com/pulsar-edit/pulsar/pull/974/commits/3d70f629534df633a4d2fb20151fa9f7b9b4e96b)
>
> …which, surprising as it sounds, has a _huge_ performance benefit.

Surprising indeed!

> [[language-sass] Add SCSS Tree-sitter grammar](https://github.com/pulsar-edit/pulsar/pull/974/commits/da1912d1a70c9a9687fddddcfc8ae98b4d49a0f9)

Sounds good to me!

> [[language-ruby] Update to latest Tree-sitter Ruby parser](https://github.com/pulsar-edit/pulsar/pull/974/commits/6d36685df38fe2252ba832d0be8e9b46ec96376c)

Also sounds good!

**Pull request merged**